### ISSUE TITLE
Unable to install swirl

### ISSUE THREAD

**Issue by bkornpob** [swirldev/swirl#327]
> install.packages("swirl", dependencies = TRUE)
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  unable to access index for repository http://cran.rstudio.com/src/contrib
Installing package into ‘C:/Users/Kornpob/Documents/R/win-library/3.2’
(as ‘lib’ is unspecified)
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  unable to access index for repository http://cran.rstudio.com/src/contrib
Warning in install.packages :
  package ‘swirl’ is not available (for R version 3.2.1)

----------------------------------------------------------------------------------------------------------------------------

I have followed suggestion from #325, #319, but still didn't work. Please help.
Thank you.

**Comment by seankross** [comment]
Thanks for looking at the other issues first. This has become a widespread issue this week and we're trying to figure out why. Here's something different to try:
```r
install.packages("devtools")
devtools::install_github("swirldev/swirl", build_vignettes = FALSE)
```

**Comment by bkornpob** [comment]
> install.packages("devtools")
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  unable to access index for repository http://cran.rstudio.com/src/contrib
Installing package into ‘C:/Users/Kornpob/Documents/R/win-library/3.2’
(as ‘lib’ is unspecified)
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  InternetOpenUrl failed: 'The HTTP redirect request failed'
Warning in install.packages :
  unable to access index for repository http://cran.rstudio.com/src/contrib
Warning in install.packages :
  package ‘devtools’ is not available (for R version 3.2.1)
--------------------------------------------------------------------------------------------------------------------------
> devtools::install_github("swirldev/swirl", build_vignettes = FALSE)
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘curl’
--------------------------------------------------------------------------------------------------------------------------

I tried both commands and they didn't work.
Besides using the command line, I also used the menu bar "Install Packages" from "Repository(CRAN,...)" and it didn't work either.

**Comment by bkornpob** [comment]
The problem is fixed as followed:

1. Install a package named "httr" <<< this will give "httr" and "curl" packages
2. Install "swirl"
